### PR TITLE
[auth0-lock] add passwordless mode class

### PR DESCRIPTION
**Pull request by himu62**
issue https://github.com/DefinitelyTyped/DefinitelyTyped/issues/27664

refs
* https://github.com/auth0/lock-passwordless/issues/118
* https://github.com/auth0/lock/blob/master/src/passwordless.js
* https://auth0.com/docs/libraries/lock#passwordless

This is my first commit for DefinitelyTyped, so please give me any advice.

## New Features

```ts
import { Auth0LockPasswordless } from "auth0-lock";

const lock: Auth0LockPasswordlessStatic = Auth0LockPasswordless(CLIENT_ID, DOMAIN, options);
```

```ts
import { Auth0Lock } from "auth0-lock";
```

----

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/auth0/lock/blob/master/src/passwordless.js
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/auth0-lock/index.d.ts [hunk 1]
```diff
@@ -223,8 +221,20 @@ interface Auth0LockStatic {
...
+interface Auth0LockPasswordlessConstructorOptions extends Auth0LockConstructorOptions {
+    passwordlessMethod?: string;
+}
+
+interface Auth0LockPasswordlessStatic extends Auth0LockCore {
+    new (clientId: string, domain: string, options?: Auth0LockPasswordlessConstructorOptions): Auth0LockPasswordlessStatic;
+}
 
 declare module "auth0-lock" {
-    export default Auth0Lock;
```
Comment (xsv24) [inline types/auth0-lock/index.d.ts:229]: 👋  Just leave the default export and add the extra exports in-order to be consistent with auth-lock libary plus that way there are no breaking changes.

https://github.com/auth0/lock
```js
import Auth0Lock from 'auth0-lock';
// OR
import { Auth0Lock } from 'auth0-lock';
import { Auth0LockPasswordless } from 'auth0-lock';
```

**Review by xsv24 — CHANGES_REQUESTED** [review]

File: types/auth0-lock/index.d.ts (hunk 1 above)
Comment (himu62) [reply types/auth0-lock/index.d.ts:229]: I added default export

**Review by xsv24 — APPROVED** [review]
Great! All approved.

**Comment by himu62** [comment]
Ready to merge

**Pull request merged**